### PR TITLE
Project 5 (PostgreSQL Database Integration)

### PR DESCRIPTION
**Pull request by Bee0933**
- used and setup `poetry` for virtual environment and dependency management
- created a `main.py` entry-point script 
- included `exceptions.py` & `utils.py` files to aid `main.py` script logic
- defined a utility class `DBUtility` with sub methods to handle project logic
-  integrated a local postgres db connection to the script
- added info and setup documentation

### CODE REVIEW

File: project-5/main.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+    def data_type_validation(self):
+        """
+        This function validates the data type
+        """
+        for num in range(len(self.df)):
+            if (isinstance(self.df['Company Name'][num], str) and
+                    isinstance(self.df['Company Link'][num], str) and
+                    isinstance(self.df['Company LinkedIn'][num], str)):
+                logging.info('Data type validated')
```
Comment (elseagle) [inline project-5/main.py:164]: What happens if validation fails ?

File: project-5/main.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+            raise ConnectToDatabaseError("cannot connect to postgres database") from e
+
+    def create_postgres_table(self) -> None:
+        """
+        This function creates the postgresql table if it does not exist in the database
+        """
+        query = """
+        CREATE TABLE IF NOT EXISTS companies (
+        id SERIAL PRIMARY KEY,
+        company_name VARCHAR(100) NOT NULL,
```
Comment (elseagle) [inline project-5/main.py]: The `company_` prefix is not need

**Review by elseagle — CHANGES_REQUESTED** [review]
Solid work! Just a little comment and we're good.

File: project-5/main.py (hunk 1 above)
Comment (Kong-Codes) [reply project-5/main.py:164]: done

File: project-5/main.py (hunk 2 above)
Comment (Kong-Codes) [reply project-5/main.py]: done

**Review by elseagle — APPROVED** [review]

**Pull request merged**